### ISSUE TITLE
Pre-commit hook is causing errors.

### ISSUE THREAD

**Issue by Brondahl** [emotion-js/emotion#914]
I actually have a bunch of issues with the pre-commit hook that makes automated changes (I assume applying Prettier recommendations?), but the majority are merely annoying, or opinions on Prettier settings, but I've discovered a case where **_these automated steps are actively breaking the codebase_**.

The file: `packages/create-emotion-server/types/index.d.ts` currently looks like this on `master`:
```
/// <reference types="node" />
// TypeScript Version: 2.3 

import { Emotion } from 'create-emotion';

export interface EmotionServer {
  extractCritical(html: string): { html: string; ids: string[]; css: string; };
  renderStylesToString(html: string): string;
  renderStylesToNodeStream(): NodeJS.ReadWriteStream;
}

export default function createEmotionServer(emotion: Emotion): EmotionServer;
```

If I try to add a space to the comment, and commit that change, then the pre-commit hook applies various automated changes, and results in a file that looks like this:

```
/// <reference types="node" />
// TypeScript Version: 2.3

import { Emotion } from 'create-emotion'

export interface EmotionServer {
  extractCritical(html: string): { html: string; ids: string[]; css: string }
  renderStylesToString(html: string): string
  renderStylesToNodeStream(): NodeJS.ReadWriteStream
}

export default function createEmotionServer(emotion: Emotion): EmotionServer
```

The main change appears to be the removal of all the semicolons.

Before the change `yarn test` runs smoothly.
After the change, but before committing `yarn test` runs smoothly.
After committing `yarn test` fails with: 
```
 RUNS  packages/styled/__tests__/styled.js
Parsing service failed to cat C:\Work\TwoGenerations\emotionPR\emotion\node_modules\@emotion\stylis\dist\stylis.esm.js, so skipping it. Exception: Sys_error("C:\\Work\\TwoGenerations\\emotionPR\\emotion\\node_modules\\@emotion\\stylis\\dist\\stylis.esm.js: No such file or directory")
Parsing service failed to cat C:\Work\TwoGenerations\emotionPR\emotion\node_modules\@emotion\stylis\dist\stylis.cjs.prod.js, so skipping it. Exception: Sys_error("C:\\Work\\TwoGenerations\\emotionPR\\emotion\\node_modules\\@emotion\\stylis\\dist\\stylis.cjs.prod.js: No such file or directory")
Parsing service failed to cat C:\Work\TwoGenerations\emotionPR\emotion\node_modules\@emotion\stylis\dist\stylis.cjs.js.flow, so skipping it. Exception: Sys_error("C:\\Work\\TwoGenerations\\emotionPR\\emotion\\node_modules\\@emotion\\stylis\\dist\\stylis.cjs.js.flow: No such file or directory")
Parsing service failed to cat C:\Work\TwoGenerations\emotionPR\emotion\node_modules\@emotion\stylis\dist\stylis.cjs.js, so skipping it. Exception: Sys_error("C:\\Work\\TwoGenerations\\emotionPR\\emotion\\node_modules\\@emotion\\stylis\\dist\\stylis.cjs.js: No such file or directory")
Parsing service failed to cat C:\Work\TwoGenerations\emotionPR\emotion\node_modules\@emotion\stylis\dist\stylis.cjs.dev.js, so skipping it. Exception: Sys_error("C:\\Work\\TwoGenerations\\emotionPR\\emotion\\node_modules\\@emotion\\stylis\\dist\\stylis.cjs.dev.js: No such file or directory")
Parsing service failed to cat C:\Work\TwoGenerations\emotionPR\emotion\node_modules\@emotion\stylis\dist\stylis.browser.cjs.js, so skipping it. Exception: Sys_error("C:\\Work\\TwoGenerations\\emotionPR\\emotion\\node_modules\\@emotion\\stylis\\dist\\stylis.browser.cjs.js: No such file or directory")
Parsing service failed to cat C:\Work\TwoGenerations\emotionPR\emotion\node_modules\@emotion\stylis\dist\index.min.js, so skipping it. Exception: Sys_error("C:\\Work\\TwoGenerations\\emotionPR\\emotion\\node_modules\\@emotion\\stylis\\dist\\index.min.js: No such file or directory")
create-emotion-server: Error: C:/Work/TwoGenerations/emotionPR/emotion/packages/create-emotion-server/types/index.d.ts:4:41
create-emotion-server: ERROR: 4:41   semicolon  Missing semicolon
create-emotion-server: ERROR: 10:50  semicolon  Missing semicolon
create-emotion-server: ERROR: 11:45  semicolon  Missing semicolon
create-emotion-server: ERROR: 12:53  semicolon  Missing semicolon
create-emotion-server: ERROR: 15:77  semicolon  Missing semicolon
create-emotion-server:     at C:\Work\TwoGenerations\emotionPR\emotion\packages\create-emotion-server\node_modules\dtslint\bin\index.js:101:19
create-emotion-server:     at Generator.next (<anonymous>)
create-emotion-server:     at fulfilled (C:\Work\TwoGenerations\emotionPR\emotion\packages\create-emotion-server\node_modules\dtslint\bin\index.js:5:58)
create-emotion-server: npm ERR! code ELIFECYCLE
create-emotion-server: npm ERR! errno 1
create-emotion-server: npm ERR! create-emotion-server@10.0.0-really-unsafe-please-do-not-use.2 test:typescript: `dtslint types`
create-emotion-server: npm ERR! Exit status 1
create-emotion-server: npm ERR!
create-emotion-server: npm ERR! Failed at the create-emotion-server@10.0.0-really-unsafe-please-do-not-use.2 test:typescript script.
create-emotion-server: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
create-emotion-server: npm ERR! A complete log of this run can be found in:
create-emotion-server: npm ERR!     C:\Users\MDM\AppData\Roaming\npm-cache\_logs\2018-10-11T08_48_51_421Z-debug.log
lerna ERR! npm run test:typescript exited 1 in 'create-emotion-server'
lerna ERR! npm run test:typescript stdout:

> create-emotion-server@10.0.0-really-unsafe-please-do-not-use.2 test:typescript C:\Work\TwoGenerations\emotionPR\emotion\packages\create-emotion-server
> dtslint types


lerna ERR! npm run test:typescript stderr:
Error: C:/Work/TwoGenerations/emotionPR/emotion/packages/create-emotion-server/types/index.d.ts:4:41
ERROR: 4:41   semicolon  Missing semicolon
ERROR: 10:50  semicolon  Missing semicolon
ERROR: 11:45  semicolon  Missing semicolon
ERROR: 12:53  semicolon  Missing semicolon
ERROR: 15:77  semicolon  Missing semicolon

    at C:\Work\TwoGenerations\emotionPR\emotion\packages\create-emotion-server\node_modules\dtslint\bin\index.js:101:19
    at Generator.next (<anonymous>)
    at fulfilled (C:\Work\TwoGenerations\emotionPR\emotion\packages\create-emotion-server\node_modules\dtslint\bin\index.js:5:58)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! create-emotion-server@10.0.0-really-unsafe-please-do-not-use.2 test:typescript: `dtslint types`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the create-emotion-server@10.0.0-really-unsafe-please-do-not-use.2 test:typescript script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\MDM\AppData\Roaming\npm-cache\_logs\2018-10-11T08_48_51_421Z-debug.log

lerna ERR! npm run test:typescript exited 1 in 'create-emotion-server'
lerna WARN complete Waiting for 14 child processes to exit. CTRL-C to exit immediately.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "test:typescript" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Full output file: 
[outputAfterGitCommitHookAutoChange.txt](https://github.com/emotion-js/emotion/files/2468241/outputAfterGitCommitHookAutoChange.txt)

Clearly the pre-commit hook should not _actively_ break the codebase :(

<hr>

 - emotion version: current master as of 11/10/2018
 - react version: (N/A: This is a bug with the library development, not its usage)
 - node version: `v10.10.1`
 - npm version: `6.4.1`
 - yarn version: `1.10.1`
 - git version: `2.19.1.windows.1`

**Comment by Brondahl** [comment]
Note that the pre-commit hook can be suppressed on the command line by using `--no-verify`: https://stackoverflow.com/questions/7230820/skip-git-commit-hooks

**Comment by Andarist** [comment]
Seems like our prettier/eslint rules conflict with some TS rules, cc @Ailrun

**Comment by Ailrun** [comment]
This will be fixed in #912.

**Comment by Brondahl** [comment]
Agreed. Cannot reproduce on current master.
